extensions/amp-instagram/1.0/instagram.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../src/preact';
+import {ContainWrapper} from '../../../src/preact/component';
+import {dict} from '../../../src/utils/object';
+import {getData} from '../../../src/event-helper';
+import {parseJson} from '../../../src/json';
+import {useLayoutEffect, useRef, useState} from '../../../src/preact';
+
+/**
+ * @param {!InstagramPropsDef} props
+ * @return {PreactDef.Renderable}
+ */
+export function Instagram({
+  shortcode,
+  captioned,
+  style,
+  title,
+  requestResize,
+  ...rest
+}) {
+  const iframeRef = useRef(null);
+  const [heightStyle, setHeightStyle] = useState(dict());
```

Comment:
Let's hoist this initial value as `const NO_HEIGHT_STYLE = dict()` up top. Otherwise we're doing a unused allocation.